spec/index.bs

```diff
@@ -4168,6 +4173,113 @@ enum GPUStoreOp {
 };
 </script>
 
+## Rasterization state ## {#render-pass-encoder-rasterization-state}
+
+The render pass encoder has several methods which affect how draw commands are rasterized to the
+attachments used by this encoder.
+
+### <dfn method for=GPURenderPassEncoder>setViewport(x, y, width, height, minDepth, maxDepth)</dfn> ### {#GPURenderPassEncoder-setViewport}
+
+<div algorithm="GPURenderPassEncoder.setViewport">
+    <strong>|this|:</strong> of type {{GPURenderPassEncoder}}.
+
+    **Arguments:**
+        - `float` x - Left side of the viewport in pixels
+        - `float` y - Top side of the viewport in pixels
+        - `float` |width| of the viewport in pixels
+        - `float` |height| of the viewport in pixels
+        - `float` |minDepth| - Minimum depth value of the viewport
+        - `float` |maxDepth| - Maximum depth value of the viewport. May be less than or equal to
+            |minDepth|.
+
+    **Returns:** void
+
+    Sets the {{GPURenderPassEncoder/[[viewport]]}} used during the rasterization stage to linearly
+    map from normalized device coordinates to viewport coordinates.
+
+    On the [=Device timeline=], the following steps occur:
+
+        - If the [$GPURenderPassEncoder.setViewport/setViewport Valid Usage$] rules are met:
+
+            - Set |this|.{{GPURenderPassEncoder/[[viewport]]}} to the extents passed to the method.
+
+    <div class=validusage dfn-for=GPURenderPassEncoder.setViewport>
+        <dfn abstract-op>setViewport Valid Usage</dfn>
+
+        1. |width| must be greater than `0`.
+        1. |height| must be greater than `0`.
+        1. |minDepth| must be greater than or equal to `0.0` and less than or equal to `1.0`.
+        1. |maxDepth| must be greater than or equal to `0.0` and less than or equal to `1.0`.
+
+    </div>
+
+    Issue: Allowed for GPUs to use fixed point or rounded viewport coordinates
+
+</div>
+
+### <dfn method for=GPURenderPassEncoder>setScissorRect(x, y, width, height)</dfn> ### {#GPURenderPassEncoder-setScissorRect}
+
+<div algorithm="GPURenderPassEncoder.setScissorRect">
+    <strong>|this|:</strong> of type {{GPURenderPassEncoder}}.
+
+    **Arguments:**
+        - {{GPUIntegerCoordinate}} x - Left side of the scissor rectangle in pixels
+        - {{GPUIntegerCoordinate}} y - Top side of the scissor rectangle in pixels
```

Comment:
We may want to avoid words like "left" and "top" in contexts where there is no fundamental physical orientation (e.g. in this case, you could be rendering into a plain texture, which has no physical orientation). In that case `x` would be the "minimum x value of the scissor rectangle".

OTOH, we could give every 2d space an orientation "by convention", i.e. say that texture x+ is to the right and y+ is down, just for the sake of making the spec easier to read/write (and any diagrams).

Not sure which is better. (@Kangz)

Reply:
Changed the text in all cases to say `minimum ___ value of...`, since that seems to be the most generally applicable.